### PR TITLE
DROID-2920 App | Tech | Introduce global subscription for profile object + get sharedSpaceCount from profile object

### PR DESCRIPTION
**Pull request by uburoiubu**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt [hunk 1]
```diff
@@ -205,4 +205,19 @@ fun SpaceViewSubscriptionContainer.isSharingLimitReached(
...
+}
+
+fun SpaceViewSubscriptionContainer.sharedSpaceCount(
+    spaceToUserPermissions: Flow<Map<Id, SpaceMemberPermissions>>
+) : Flow<Int> {
+    return combine(
+        observe(),
+        spaceToUserPermissions
+    ) { spaceViews, permissions ->
+        spaceViews.count { spaceView ->
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt:217]: please add Timber.d("Total shared space count: $count") and .catch {  Timber.e(exception, "Error calculating shared space count") }

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    class Default @Inject constructor(
+        private val scope: CoroutineScope,
+        private val configStorage: ConfigStorage,
+        private val awaitAccountStartManager: AwaitAccountStartManager,
+        private val container: StorelessSubscriptionContainer,
+        private val dispatchers: AppCoroutineDispatchers
+    ) : ProfileSubscriptionManager {
+
+        private val state = MutableStateFlow<ObjectWrapper.Basic?>(null)
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:37]: better to use ProfileSubscriptionManagerState

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                            searchParams = StoreSearchByIdsParams(
+                                subscription = GLOBAL_PROFILE_SUBSCRIPTION,
+                                space = SpaceId(config.techSpace),
+                                keys = listOf(
+                                    Relations.ID,
+                                    Relations.NAME,
+                                    Relations.ICON_EMOJI,
+                                    Relations.ICON_IMAGE,
+                                    Relations.ICON_OPTION,
+                                    Relations.SHARED_SPACES_LIMIT
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:59]: we have spaceViewKeys for that kind of cases

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                                ),
+                                targets = listOf(config.profile)
+                            )
+                        ).map {
+                            it.firstOrNull()
+                        }
+                    }
+                    .flowOn(dispatchers.io)
```
Comment (konstantiniiv) [inline domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:67]: catch {}

**Review by konstantiniiv — APPROVED** [review]

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt (hunk 3 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:59]: unfortunately, it's not available in domain module. need to move it somewhere else. need to discuss that.

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt (hunk 4 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:67]: fixed

File: domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt (hunk 2 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt:37]: will improve it in the next prs

File: domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt (hunk 1 above)
Comment (uburoiubu) [reply domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt:217]: done

**Pull request merged**